dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -238,10 +238,13 @@ T saveVersion(User user, Long entryId, T entry, U version, Set<SourceFile> versi
         validatedVersion.setParent(entry);
         long l = getVersionDAO().create(validatedVersion);
         entry.getWorkflowVersions().add(getVersionDAO().findById(l));
-        entry.checkAndSetDefaultVersion(validatedVersion.getName());
+        // Only set if the default version isn't already there
+        if (entry.getDefaultVersion() == null) {
```

Comment:
This is changing from the default always being the most recent to being "sticky". 
Thoughts?

Reply:
I think the problem was that the user keeps refreshing the entire organization and the UI would change the implicit default version to the most recent one (which they did not want).  This will set the default version to the most recent one once.  If it's the wrong one they can just change it to whatever

Reply:
seems reasonable, pinging @coverbeck @wshands for thoughts

Reply:
Agreed, the implicit default version has led to confusion with Terra team.

Reply:
I agree